plugins/mongodb-adapter/src/main/java/org/polypheny/db/adapter/mongodb/rules/MongoSort.java

```diff
@@ -0,0 +1,112 @@
+ */
+
+package org.polypheny.db.adapter.mongodb.rules;
+
+
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.annotations.Nullable;
+import org.polypheny.db.adapter.mongodb.MongoAlg;
+import org.polypheny.db.algebra.AlgCollation;
+import org.polypheny.db.algebra.AlgFieldCollation;
+import org.polypheny.db.algebra.AlgNode;
+import org.polypheny.db.algebra.core.Sort;
+import org.polypheny.db.algebra.metadata.AlgMetadataQuery;
+import org.polypheny.db.algebra.type.AlgDataTypeField;
+import org.polypheny.db.plan.AlgCluster;
+import org.polypheny.db.plan.AlgOptCost;
+import org.polypheny.db.plan.AlgPlanner;
+import org.polypheny.db.plan.AlgTraitSet;
+import org.polypheny.db.rex.RexLiteral;
+import org.polypheny.db.rex.RexNode;
+import org.polypheny.db.util.Util;
+
+
+/**
+ * Implementation of {@link Sort} relational expression in MongoDB.
+ */
+public class MongoSort extends Sort implements MongoAlg {
+
+    public MongoSort( AlgCluster cluster, AlgTraitSet traitSet, AlgNode child, AlgCollation collation, @Nullable List<RexNode> fieldExpr, RexNode offset, RexNode fetch ) {
+        super( cluster, traitSet, child, collation, fieldExpr, offset, fetch );
+        assert getConvention() == CONVENTION;
+        assert getConvention() == child.getConvention();
+    }
+
+
+    @Override
+    public Sort copy( AlgTraitSet traitSet, AlgNode newInput, AlgCollation newCollation, ImmutableList<RexNode> fieldExps, RexNode offset, RexNode fetch ) {
+        return new MongoSort( getCluster(), traitSet, newInput, collation, fieldExps, offset, fetch );
+    }
+
+
+    @Override
+    public AlgOptCost computeSelfCost( AlgPlanner planner, AlgMetadataQuery mq ) {
+        return super.computeSelfCost( planner, mq ).multiplyBy( 0.05 );
+    }
+
+
+    @Override
+    public void implement( Implementor implementor ) {
+        implementor.visitChild( 0, getInput() );
+        if ( !collation.getFieldCollations().isEmpty() ) {
+            final List<String> keys = new ArrayList<>();
+            final List<AlgDataTypeField> fields = getTupleType().getFields();
+            for ( AlgFieldCollation fieldCollation : collation.getFieldCollations() ) {
+                // we can only sort by field not by db.collection.field
+                String name =
+                        fields.get( fieldCollation.getFieldIndex() ).getName();
+                String[] splits = name.split( "\\." );
+                name = splits[splits.length - 1];
+                name = MongoRules.adjustName( name );
+                keys.add( name + ": " + direction( fieldCollation ) );
+                if ( false ) {
+                    // TODO: NULLS FIRST and NULLS LAST
+                    switch ( fieldCollation.nullDirection ) {
+                        case FIRST:
+                            break;
+                        case LAST:
+                            break;
+                    }
+                }
```

Comment:
There seems some implementation missing

Reply:
MongoDB does not allow to specify which position (first/last) null values take in sorting. It always handles them as the "smallest". I added a handling to block the push-down in those specific cases and removed this.